SmartCar/Arduino/sketch_main/sketch_main.ino

```diff
@@ -26,26 +70,87 @@ void loop() {
         case 'd':
-            motor.write(input.substring(1, 4).toInt());
-            break;
-      
-        //Turns a LED on when the app connects
-        case 'L':
-            digitalWrite(12, HIGH);
+            velocity = input.substring(1, 4).toInt();
+            if ((!flame && !obstacle) || velocity >= 90){
+                motor.write(velocity);
+            }/* else{
+                motor.write(START_POSITION_MOTOR_SERVO); //REMOVE THIS??????????????????????????????????????????
+                motor.write(110);
+                delay(5);
+                motor.write(START_POSITION_MOTOR_SERVO);
+                collision_delay = 0;
+            }*/
             break;
-          
+     
         default:
-            motor.write(90);
-            steer.write(45);
+            motor.write(START_POSITION_MOTOR_SERVO);
+            steer.write(START_POSITION_STEER_SERVO);
             break;
         }
-    
         //Resets the it so the next command can be recorded
         input = "";
         completeCommand = false;
     }
+
+    //Reads the value from the flame sensor
+    flame_pin = digitalRead (FLAME_SENSOR_PIN) ;
+
+    //If the flame sensor sees a flame it writes a command that says it sees a flame
+    if (flame_pin == HIGH && !flame && ++flame_delay >= 500){
+      Serial.write("f001\n");
+      flame = true;
+      digitalWrite(LED_PIN, HIGH);
+
+    //when the flame disapears it sends a command that it stopped seing a flame
+    }else if(flame && flame_pin == LOW){
+      Serial.write("f000\n");
+      flame = false;
+      digitalWrite(LED_PIN, LOW);
+      flame_delay = 0;
+    }
+    /*
+    //Moves the servo connected to the ultrasonic sensor
+    collisionDirection ? collisionServoPosition-- : collisionServoPosition++;
+    collisionServo.write(collisionServoPosition);
+
+    //Switches the direction of the collision servo
+    if (collisionServoPosition >= 90){
+        collisionDirection = true;
+    }else if (collisionServoPosition <= 10){
+        collisionDirection = false;
+    }  
+    */
+    //Pings the ultrasonic sensor
+    range_string = sonar.ping_cm();
+    range = range_string.toInt();
+    range_string = "c" + range_string + '\n';
+    if (lastDistance != range && ++distance_Delay >= 100){
+       // Serial.write(range);
+        Serial.print(range_string);
+        lastDistance = range;
+        distance_Delay = 0;
+    }
+
+    //Allows the car to drive backwards if an obstacle is in from of the car
+    if (velocity < 90){
+        if (((range != 0 && range <= 50) && ++collision_delay >= 250) || flame){
+            if (range > 20){
+                motor.write(velocity +(50 - range)/2);
+            } else {
+                obstacle = true;
+                collision_delay = 0;
+    
+                motor.write(START_POSITION_MOTOR_SERVO +30);
+                delay(300);
+                motor.write(START_POSITION_MOTOR_SERVO);
+            }
+        } else if (obstacle){
```

Comment:
What's the purpose of this else if? I get that it stops it when it detects an object. Bit confusing.